pkg/controllermanager/controller/shoot/shoot_maintenance_control.go

```diff
@@ -69,94 +73,92 @@ func (c *Controller) shootMaintenanceDelete(obj interface{}) {
 	}
 	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
 	if err != nil {
+		c.log.Error(err, "Couldn't get key for object", "object", obj)
 		return
 	}
 	c.shootMaintenanceQueue.Done(key)
 }
 
 // NewShootMaintenanceReconciler creates a new instance of a reconciler which maintains Shoots.
-func NewShootMaintenanceReconciler(l logrus.FieldLogger, gardenClient client.Client, config config.ShootMaintenanceControllerConfiguration, recorder record.EventRecorder) reconcile.Reconciler {
+func NewShootMaintenanceReconciler(gardenClient client.Client, config config.ShootMaintenanceControllerConfiguration, recorder record.EventRecorder) reconcile.Reconciler {
 	return &shootMaintenanceReconciler{
-		logger:       l,
 		gardenClient: gardenClient,
 		config:       config,
 		recorder:     recorder,
 	}
 }
 
 type shootMaintenanceReconciler struct {
-	logger       logrus.FieldLogger
 	gardenClient client.Client
 	config       config.ShootMaintenanceControllerConfiguration
 	recorder     record.EventRecorder
 }
 
 func (r *shootMaintenanceReconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	log := logf.FromContext(ctx)
+
 	shoot := &gardencorev1beta1.Shoot{}
 	if err := r.gardenClient.Get(ctx, request.NamespacedName, shoot); err != nil {
 		if apierrors.IsNotFound(err) {
-			r.logger.Infof("Object %q is gone, stop reconciling: %v", request.Name, err)
+			log.Info("Object is gone, stop reconciling")
 			return reconcile.Result{}, nil
 		}
-		r.logger.Infof("Unable to retrieve object %q from store: %v", request.Name, err)
-		return reconcile.Result{}, err
+		return reconcile.Result{}, fmt.Errorf("error retrieving object from store: %w", err)
 	}
 
 	if shoot.DeletionTimestamp != nil {
-		r.logger.Debug("[SHOOT MAINTENANCE] - skipping because Shoot is marked as to be deleted")
+		log.V(1).Info("Skipping Shoot because it is marked for deletion")
 		return reconcile.Result{}, nil
 	}
 
-	requeueAfter := requeueAfterDuration(shoot)
+	requeueAfter, nextMaintenance := requeueAfterDuration(shoot)
 
 	if !mustMaintainNow(shoot) {
-		logger.Logger.Infof("[SHOOT MAINTENANCE] %s/%s - skipping because Shoot must not be maintained now.", shoot.Namespace, shoot.Name)
+		log.V(1).Info("Skipping Shoot because it doesn't need to be maintained now")
+		log.V(1).Info("Scheduled next maintenance for Shoot", "duration", requeueAfter.Round(time.Minute), "nextMaintenance", nextMaintenance.Round(time.Minute))
```

Comment:
Why `V(1)`? This was `Info` before

Reply:
I think, logging the scheduled next maintenance for every shoot is pretty noisy. Hence, increased the log level to debug